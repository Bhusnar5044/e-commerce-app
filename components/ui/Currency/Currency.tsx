('use client');

import { memo, useEffect, useState } from 'react';

const formatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
});

interface CurrencyProps {
    value?: string | number;
}

export const Currency: React.FC<CurrencyProps> = memo(({ value = 0 }) => {
    const [isMounted, setIsMounted] = useState(false);

    useEffect(() => {
        setIsMounted(true);
    }, []);

    if (!isMounted) {
        return null;
    }

    return <div className="font-semibold">{formatter.format(Number(value))}</div>;
});

Currency.displayName = 'Currency';
