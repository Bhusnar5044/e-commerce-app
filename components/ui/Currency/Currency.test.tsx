import { render } from '@test-utils';
import { Currency } from './Currency';

describe('Currency', () => {
    it('should render properly', () => {
        const { container } = render(<Currency />);
        expect(container).toMatchSnapshot();
    });
});
