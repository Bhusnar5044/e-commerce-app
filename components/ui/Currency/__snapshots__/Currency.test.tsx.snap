// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency should render properly 1`] = `
<div>
  <div
    class="font-semibold"
  >
    $0.00
  </div>
</div>
`;
